### PR TITLE
Handle the StatusUpdated event.

### PR DESCRIPTION
**Pull request by LucWollants**
### Added

- Added a StatusUpdatedProjector to handle the StatusUpdated event on either an event or place.

---

Ticket: https://jira.uitdatabank.be/browse/III-3656

### CODE REVIEW

File: src/Offer/ReadModel/JSONLD/StatusUpdatedProjector.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        $json['status'] = $statusUpdated->getStatus()->serialize();
+
+        if (!empty($json['subEvent'])) {
+            $nrOfSubEvents = count($json['subEvent']);
+            for ($subEventIndex = 0; $subEventIndex < $nrOfSubEvents; $subEventIndex++) {
+                $json['subEvent'][$subEventIndex]['status'] = $statusUpdated->getStatus()->serialize();
+            }
+        }
+
+        $this->eventRepository->save($jsonDocument->withAssocBody($json));
```
Comment (jdrieghe) [inline src/Offer/ReadModel/JSONLD/StatusUpdatedProjector.php:77]: Should this use the passed `$documentRepository`?

**Review by jdrieghe — CHANGES_REQUESTED** [review]
I think we're using the wrong repository to save the document.

In general, I'm not a big fan of the semi-abstraction used here.
I would much rather inject an `OfferRepository` and modify the offer without caring about its type. If that's not possible, I would create separate projectors. Right now, it's somewhere in the middle and can become very confusing and error-prone when another implementation of that abstraction enters the equation.

File: src/Offer/ReadModel/JSONLD/StatusUpdatedProjector.php (hunk 1 above)
Comment (LucWollants) [reply src/Offer/ReadModel/JSONLD/StatusUpdatedProjector.php:77]: Indeed the wrong repository!

**Comment by LucWollants** [comment]
> I would much rather inject an `OfferRepository` and modify the offer without caring about its type.
That is certainly possible, but that new `OfferRepository` would have the logic to `load` and `save` from the correct repo. So for example it would first `load` from the events repo, if nothing is found it would load from the places repo.

**Comment by jdrieghe** [comment]
@LucWollants That's exactly what I mean. Now, this logic is spread across potentially multiple projectors when we could easily abstract that away to just one place. Even saving can be done based on the type in the read model.

**Comment by LucWollants** [comment]
The `JsonDocument` has no real type. So when doing a `save` it can become tricky to determine the correct repo because what if it's a new event or place to save?

**Review by jdrieghe — APPROVED** [review]

**Comment by jdrieghe** [comment]
Looks great, @LucWollants. I'm happy we can now safely refactor away all the duplicate projectors as well.

**Pull request merged**